### PR TITLE
🎉 add spotify listening to field

### LINKED ISSUE #7: Add spotify listening to field
**Is your feature request related to a problem? Please describe.**
It will be good to have a section in the readme that shows the current song I'm listening to on Spotify.

**Describe the solution you'd like**
This [project](https://github.com/kittinan/spotify-github-profile) can be integrated.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by kairavkkp**
resolves #7

### CODE REVIEW

File: src/components/Field/SpotifyListeningTo.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import { FieldProps } from '.';
+import { globalContext } from '../../context/GlobalContextProvider';
+
+export enum STATS_ALIGNMENT {
+    LEFT = 'left',
+    CENTRE = 'center',
+    RIGHT = 'right',
+}
+
+export enum VARIANT {
```
Comment (rishavanand) [inline src/components/Field/SpotifyListeningTo.tsx:17]: Let's remove this as it's not used

File: src/components/Field/SpotifyListeningTo.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            <Menu.Item key="2" onClick={() => changeAlignment(STATS_ALIGNMENT.CENTRE)}>
+                Centre
+            </Menu.Item>
+            <Menu.Item key="3" onClick={() => changeAlignment(STATS_ALIGNMENT.RIGHT)}>
+                Right
+            </Menu.Item>
+        </Menu>
+    );
+
+    const variantMenu = (
```
Comment (rishavanand) [inline src/components/Field/SpotifyListeningTo.tsx:129]: Can also remove variant menu

File: src/components/Field/SpotifyListeningTo.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    };
+
+    const changeAlignment = (alignment: typeof localSpotifyListeningProps.options.alignment) => {
+        const localProps = { ...localSpotifyListeningProps };
+        if (!localProps.options) localProps.options = {};
+        localProps.options.alignment = alignment;
+        modifyField(localProps);
+    };
+
+    const changeVariant = (variant: VARIANT) => {
```
Comment (rishavanand) [inline src/components/Field/SpotifyListeningTo.tsx:108]: Can also remove change variant method

File: src/components/Field/SpotifyListeningTo.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    RIGHT = 'right',
+}
+
+export enum VARIANT {
+    LANGUAGE_STATS = 'languages',
+    ACTIVITY_STATS = 'activity',
+}
+
+export interface SpotifyListeningToOptions {
+    height?: string;
```
Comment (rishavanand) [inline src/components/Field/SpotifyListeningTo.tsx:23]: Can remove height, width and variant

File: src/components/Field/SpotifyListeningTo.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+import { Input, Row, Col, Button, Dropdown, Menu, Form, Switch } from 'antd';
+import { DownOutlined } from '@ant-design/icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import styles from '../../styles/fields.module.scss';
+import { faAlignLeft, faExpandArrowsAlt } from '@fortawesome/free-solid-svg-icons';
+import { faSpotify } from '@fortawesome/free-brands-svg-icons';
+import { FieldProps } from '.';
+import { globalContext } from '../../context/GlobalContextProvider';
+
+export enum STATS_ALIGNMENT {
```
Comment (rishavanand) [inline src/components/Field/SpotifyListeningTo.tsx:11]: Can replace all STATS_ALIGNMENT with ALIGNMENT

File: src/components/Field/SpotifyListeningTo.tsx [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+        return `</div>`;
+    else return '';
+};
+
+export const generateSpotifyListeningToMarkdown = ({ data, options }: SpotifyListeningToProps) => {
+    const markdown = data.spotifyMarkdown
+        ? data.spotifyMarkdown
+        : '[spotify-github-profile](https://rishavanand.github.io/static/images/spotify-readme-example.svg)';
+    const svgLink = markdown.substring(markdown.indexOf('(') + 1, markdown.indexOf(')'));
+    console.log(svgLink);
```
Comment (rishavanand) [inline src/components/Field/SpotifyListeningTo.tsx:53]: console log not needed in production

File: src/components/Field/SpotifyListeningTo.tsx (hunk 1 above)
Comment (kairavkkp) [reply src/components/Field/SpotifyListeningTo.tsx]: Solved in the latest commit.

File: src/components/Field/SpotifyListeningTo.tsx (hunk 2 above)
Comment (kairavkkp) [reply src/components/Field/SpotifyListeningTo.tsx]: Solved in the latest commit.

File: src/components/Field/SpotifyListeningTo.tsx (hunk 3 above)
Comment (kairavkkp) [reply src/components/Field/SpotifyListeningTo.tsx]: Solved in the latest commit.

File: src/components/Field/SpotifyListeningTo.tsx (hunk 4 above)
Comment (kairavkkp) [reply src/components/Field/SpotifyListeningTo.tsx]: Solved in the latest commit.

File: src/components/Field/SpotifyListeningTo.tsx (hunk 5 above)
Comment (kairavkkp) [reply src/components/Field/SpotifyListeningTo.tsx]: Solved in the latest commit.

File: src/components/Field/SpotifyListeningTo.tsx (hunk 6 above)
Comment (kairavkkp) [reply src/components/Field/SpotifyListeningTo.tsx]: Solved in the latest commit.

**Pull request merged**